packages/data-transport-layer/src/utils/eth-tx.ts

```diff
@@ -1,9 +1,18 @@
 /* Imports: External */
 import { ethers } from 'ethers'
 
+// This is expected to return 0 or 1
 export const parseSignatureVParam = (
   v: number | ethers.BigNumber,
   chainId: number
 ): number => {
-  return ethers.BigNumber.from(v).toNumber() - 2 * chainId - 35
+  if (typeof v === 'object') {
+    v = ethers.BigNumber.from(v).toNumber()
+  }
+  // Handle unprotected transactions
+  if (v === 27 || v === 28) {
+    return v - 27
+  }
+  // Handle EIP155 transactions
+  return v - 2 * chainId - 35
```

Comment:
why is this transform necessary? wondering what we do to this `v` parameter that's different from i assume an l1 eth transaction

Reply:
The chainid is used to allow replay protection, see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md